src/main/java/Duke.java

```diff
@@ -1,10 +1,254 @@
+            }
+        } else {
+            System.out.println("There is no tasks");
+        }
+    }
+
+    private static void doDelete(ArrayList<Task> tasks, String[] wordList) throws DukeException {
+        if (wordList.length != 2) {
+            throw new DukeException("☹ OOPS!!! Wrong number of arguments for delete");
+        }
+        try {
+            int index = Integer.parseInt(wordList[1]) - 1;
+            if (index + 1 > tasks.size()) {
+                throw new DukeException("☹ OOPS!!! index out of bounds");
+            } else {
+                System.out.println("Noted. I've removed this task:");
+                System.out.println(tasks.get(index));
+                tasks.remove(index);
+                System.out.printf("Now you have %d tasks in the list\n", tasks.size());
+            }
+
+        } catch (Exception exception) {
+            System.out.println(exception);
+        }
+        return;
+    }
+
+    private static void processInput(String line, ArrayList<Task> tasks, String[] wordList, String command) throws
+            DukeException, IOException {
+        switch (command) {
+        case "list":
+            printList(tasks, wordList);
+            break;
+        case "unmark":
+            doUnmark(tasks, wordList);
+            doSave(tasks);
+            break;
+        case "mark":
+            doMark(tasks, wordList);
+            doSave(tasks);
+            break;
+        case "deadline":
+            doDeadline(line, tasks);
+            doSave(tasks);
+            break;
+        case "todo":
+            doTodo(line, tasks);
+            doSave(tasks);
+            break;
+        case "event":
+            doEvent(line, tasks);
+            doSave(tasks);
+            break;
+        case "delete":
+            doDelete(tasks, wordList);
+            doSave(tasks);
+            break;
+        default:
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    private static void doSave(ArrayList<Task> tasks) throws DukeException, IOException {
+        File file = new File(LOAD_PATH);
+        if (!file.getParentFile().exists()) {
+            file.getParentFile().mkdirs();
+        }
+        if (!file.exists()) {
+            file.createNewFile();
+        }
+        FileWriter fileWriter = new FileWriter(file);
+        for (Task task : tasks) {
+            fileWriter.write(task.toString());
+            fileWriter.write(System.lineSeparator());
+        }
+        fileWriter.close();
+    }
+
+    private static void doLoad(ArrayList<Task> tasks, File file) throws DukeException {
+        try {
+            if (file.exists()) {
+                System.out.println("Welcome back!");
+                System.out.println("Here are the tasks from last time:");
+                Scanner readFile = new Scanner(file);
+                while (readFile.hasNextLine()) {
```

Comment:
Perhaps avoid deep nesting where you can. You can refactor the code in the inner nest to a method and call it instead. [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)